### PR TITLE
Add text screenshots

### PR DESCRIPTION
**Pull request by djansen-redhat**

### CODE REVIEW

**Comment by djansen-redhat** [comment]
@zanata-jenkins retest this please

**Comment by djansen-redhat** [comment]
@zanata-jenkins retest this please

**Comment by djansen-redhat** [comment]
@zanata-jenkins retest this please

**Comment by definite** [comment]
@zanata-jenkins retest this please

**Comment by djansen-redhat** [comment]
@seanf would you be interested/have the capacity to review this one?

File: functional-test/src/main/java/org/zanata/page/AbstractPage.java [hunk 1]
```diff
@@ -484,6 +554,10 @@ public String getHtmlSource(WebElement webElement) {
                 "return arguments[0].innerHTML;", webElement);
     }
 
+    public void triggerScreenshot(String tag) {
+        WebElementUtil.triggerScreenshot(tag);
+    }
+
     // TODO: Deprecated functions, remove after 3.7
```
Comment (seanf) [inline functional-test/src/main/java/org/zanata/page/AbstractPage.java]: Is now a good time to delete the deprecated functions?

File: functional-test/src/main/java/org/zanata/page/AbstractPage.java [hunk 2]
```diff
@@ -356,6 +362,70 @@ public void clickElement(final WebElement element) {
...
+            enterTextAction.sendKeys(text).perform();
+        }
+
+        waitForAMoment().withMessage("Text equal to entered")
+                .until(new Predicate<WebDriver>() {
+                    @Override
+                    public boolean apply(WebDriver input) {
+                        if (!check) {
+                            log.info("Not checking text entered");
+                            return true;
```
Comment (seanf) [inline functional-test/src/main/java/org/zanata/page/AbstractPage.java]: Shouldn't we skip this whole `waitForAMoment` section if `check` is false?

**Comment by seanf** [comment]
@djansen-redhat All looks okay otherwise.

File: functional-test/src/main/java/org/zanata/page/AbstractPage.java (hunk 1 above)
Comment (djansen-redhat) [inline functional-test/src/main/java/org/zanata/page/AbstractPage.java]: Maybe in a separate commit?

File: functional-test/src/main/java/org/zanata/page/AbstractPage.java (hunk 2 above)
Comment (djansen-redhat) [inline functional-test/src/main/java/org/zanata/page/AbstractPage.java]: Good point, will fix.

File: functional-test/src/main/java/org/zanata/page/AbstractPage.java (hunk 1 above)
Comment (seanf) [inline functional-test/src/main/java/org/zanata/page/AbstractPage.java]: Sure, just thought I'd mention it.

On 3 September 2015 at 09:32, Damian Jansen <notifications@github.com>
wrote:

> In functional-test/src/main/java/org/zanata/page/AbstractPage.java
> <https://github.com/zanata/zanata-server/pull/937#discussion_r38598335>:
>
> > @@ -484,6 +554,10 @@ public String getHtmlSource(WebElement webElement) {
> >                  "return arguments[0].innerHTML;", webElement);
> >      }
> >
> > +    public void triggerScreenshot(String tag) {
> > +        WebElementUtil.triggerScreenshot(tag);
> > +    }
> > +
> >      // TODO: Deprecated functions, remove after 3.7
>
> Maybe in a separate commit?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/937/files#r38598335>.
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

**Comment by seanf** [comment]
@djansen-redhat Did you amend the commit?  Makes it hard to review the new changes.

**Comment by djansen-redhat** [comment]
@seanf I did.  All that's done is moved the if(check) to outside the waitFor.
I'll keep that in mind for next time.

**Comment by seanf** [comment]
Okay, thanks.

:+1:

**Comment by djansen-redhat** [comment]
@zanata-jenkins retest this please

**Comment by djansen-redhat** [comment]
@zanata-jenkins retest this please

**Comment by djansen-redhat** [comment]
:white_check_mark: Tested by Jenkins

**Comment by djansen-redhat** [comment]
Merged in 15ab7d835a3b052151a9079dd43e4027348c5abc

**Pull request closed without merging**